### PR TITLE
Modify validation of listen address to support IPv4 and IPv6

### PR DESCRIPTION
**Pull request by dougwettlaufer**

### CODE REVIEW

File: stargate-starter/src/main/java/io/stargate/starter/Starter.java [hunk 1]
```diff
@@ -269,6 +266,10 @@ protected void setStargateProperties() {
       throw new IllegalArgumentException("--cluster-version must be a number");
     }
 
+    if (!InetAddressValidator.getInstance().isValid(listenHostStr)) {
```
Comment (dougwettlaufer) [inline stargate-starter/src/main/java/io/stargate/starter/Starter.java:269]: Pulling in Apache Commons validator because IPv6 regexes are gnarly.

**Review by dimas-b — APPROVED** [review]
It would be nice to add integration tests for IPv6... perhaps in another PR. WDYT?

File: stargate-starter/src/test/java/io/stargate/starter/StarterTest.java [hunk 2]
```diff
@@ -171,4 +171,69 @@ void testDeveloperModeSetsDefaultSeedsAndSnitch() {
...
   }
+
+  @Test
+  void testSetStargatePropertiesWithIPv4ListenHost() {
+    Starter starter = new Starter();
+    starter.simpleSnitch = true;
+    starter.seedList = Arrays.asList("127.0.0.1", "127.0.0.2");
+    starter.listenHostStr = "127.0.0.1";
+    starter.clusterName = "foo";
+    starter.version = "3.11";
```
Comment (zznate) [inline stargate-starter/src/test/java/io/stargate/starter/StarterTest.java:182]: Readibility nit: collapse into a private method to remove copypasta and make tests read a bit quicker?

File: stargate-starter/src/test/java/io/stargate/starter/StarterTest.java [hunk 3]
```diff
@@ -156,19 +140,74 @@ void testSeedsNotPresentThrows() {
...
+    RuntimeException thrown =
+        assertThrows(
+            IllegalArgumentException.class,
+            starter::setStargateProperties,
+            "Expected setStargateProperties() to throw RuntimeException");
+
+    assertThat(thrown.getMessage()).isEqualTo("--listen must be a valid IPv4 or IPv6 address");
+  }
+
+  private Starter setupDefaultStarter() {
```
Comment (dimas-b) [inline stargate-starter/src/test/java/io/stargate/starter/StarterTest.java:204]: Could be made `@BeforeEach`, I guess, to save calling it in each test method.

Comment (dougwettlaufer) [reply stargate-starter/src/test/java/io/stargate/starter/StarterTest.java:204]: True

**Review by dimas-b — APPROVED** [review]

**Pull request merged**